roles/openstack/tasks/server.yaml

```diff
@@ -0,0 +1,141 @@
+---
+# (c) Copyright IBM Corporation
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Extract OS info from image name of server {{ server_config.name }}
+  ansible.builtin.set_fact:
+    _neerali_openstack_os_data: >-
+      {{
+        {
+          'type': server_config.image.split('-')[0] | lower,
+          'version': server_config.image.split('-')[1].split('.')[0:2] | join('.')
+        }
+      }}
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+
+- name: Extract cloud-init yaml for server {{ server_config.name }}
+  ansible.builtin.set_fact:
+    _neerali_openstack_cloud_init: "{{ server_config.cloud_init | from_yaml }}"
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+
+- name: Create empty list for network nics to reset network nics for every server
+  ansible.builtin.set_fact:
+    _neerali_openstack_network_nics: []
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+
+- name: Generate nics payload for server {{ server_config.name }}
+  ansible.builtin.set_fact:
+    _neerali_openstack_network_nics: >-
+      {{
+        _neerali_openstack_network_nics +
+        [
+          {
+            'net-name' : network_name
+          }
+        ]
+      }}
+  loop: "{{ server_config.networks }}"
+  loop_control:
+    loop_var: network_name
+  when:
+    - server_config is defined
+    - server_config.driver == "openstack"
+    - action is defined
+    - action == "create"
+    - server_config.networks is defined
```

Comment:
The set of servers held by `server_config` is already filter and will have nodes having `driver == openstack`. I beleive that can removed. 

we can fail hard if `server_config` is not defined
action can be rewritten as `action | default('') == 'create'

These two tasks use a similar check hence we can combine them into a `block`

Reply:
ok understood. Since we are filtering the systems layout in `main.yaml` and moving to `server.yaml` only if `server_config` is defined, we can ignore that duplicate check in `server.yaml`.